examples/heat_equation.jl

```diff
@@ -0,0 +1,36 @@
+# # Heat Equation
+#
+# This example demonstrates how to combine `OrdinaryDiffEq` with `DiffEqOperators` to solve a time dependent PDE.
+# We consider the heat equation on the unit interval, with dirichlet boundary conditions:
+# ∂ₜu = Δu
+# u(x=0,t)  = a
+# u(x=1,t)  = b
+# u(x, t=0) = u₀(x)
+#
+# For `a = b = 0` and `u₀(x) = sin(2πx)` a solution is given by:
+u_analytic(x, t) = sin(2*π*x) * exp(-t*(2*π)^2)
+#
+# We want to reproduce it numerically
+#
+using DiffEqOperators, OrdinaryDiffEq
+
+nknots = 100
+h = 1.0/(nknots+1)
+knots = range(h, step=h, length=nknots)
+ord_deriv = 2
+ord_approx = 2
+
+const Δ = CenteredDifference(ord_deriv, ord_approx, h, nknots)
+const bc = Dirichlet0BC(Float64)
+
+t0 = 0.0
+t1 = 0.03
+u0 = u_analytic.(knots, t0)
+
+step(u,p,t) = Δ*bc*u
+prob = ODEProblem(step, u0, (t0, t1))
+alg = KenCarp4(autodiff=false)
```

Comment:
why autodiff=false?

Reply:
If I turn it to true, I get:
```julia
MethodError: no method matching *(::GhostDerivativeOperator{Float64,DerivativeOpera
tor{Float64,1,false,Float64,StaticArrays.SArray{Tuple{3},Float64,1,3},StaticArrays.SArray{Tuple{0},St
aticArrays.SArray{Tuple{4},Float64,1,4},1,0},Nothing,Nothing},RobinBC{Float64,Array{Float64,1}}}, ::A
rray{ForwardDiff.Dual{ForwardDiff.Tag{DiffEqDiffTools.UDerivativeWrapper{ODEFunction{false,typeof(ste
p),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing
,Nothing},Float64,DiffEqBase.NullParameters},Float64},Float64,12},1})
```

Reply:
Try https://github.com/JuliaDiffEq/DiffEqOperators.jl/pull/193